testnet/stacks-node/src/nakamoto_node/miner.rs

```diff
@@ -265,8 +264,17 @@ impl BlockMinerThread {
         let stackerdb_contracts = stackerdbs
             .get_stackerdb_contract_ids()
             .expect("FATAL: could not get the stacker DB contract ids");
-        // TODO: get this directly instead of this jankiness when .signers is a boot contract
-        let signers_contract_id = boot_code_id(SIGNERS_NAME, self.config.is_mainnet());
+
+        let reward_cycle = u32::try_from(
```

Comment:
Similar to my other comment. I think reward cycle should remain a u64. We should just do u32::try_from on (reward_cycle % 2)